### PR TITLE
[PDR-908] Create a repeatable unique id for all biobank order and sample records.

### PR DESCRIPTION
**Pull request by robabram**
## Description of changes/additions
Create a repeatable unique record primary key for every biobank order and sample record.

## Tests
- [x] unit tests

### CODE REVIEW

File: tests/helpers/unittest_base.py [hunk 1]
```diff
@@ -502,7 +502,7 @@ def setUp(self, with_data=True, with_consent_codes=False) -> None:
         self.addCleanup(logger.removeHandler, stream_handler)
 
         # Change this to logging.ERROR when you want to see API server errors.
-        logger.setLevel(logging.CRITICAL)
+        logger.setLevel(logging.ERROR)
```
Comment (pegbertsch) [inline tests/helpers/unittest_base.py]: need to revert this to CRITICAL?

Comment (wangy70) [inline tests/helpers/unittest_base.py]: Do you intend to submit this change? Or by accident?

File: rdr_service/resource/generators/participant.py [hunk 2]
```diff
@@ -947,7 +948,18 @@ def _make_sample_dict_from_row(bss=None, bos=None):
...
 
+            # Create a unique repeatable primary key value for each biobank sample record.
+            test_id = BIOBANK_UNIQUE_TEST_IDS[test] if test in BIOBANK_UNIQUE_TEST_IDS else '99'
+            if bo_pk:
+                # From known biobank orders
+                id_ = int(f'{bo_pk}{test_id}')
+            else:
+                # For unknown biobank orders, use participant_id + 99 + test id.
+                # All tests here will be grouped together under this 99 id.
+                id_ = int(f'{bss.participant_id}99{test_id}')
```
Comment (pegbertsch) [inline rdr_service/resource/generators/participant.py]: Have we verified if there are any cases where a participant has more than one `biobank_stored_sample` row for the same test type, neither of which has a `biobank_order_id`?  Would that break the key uniqueness?

**Review by pegbertsch — COMMENTED** [review]
Believe the unittest logging level change needs reverting before this is merged.

File: tests/helpers/unittest_base.py (hunk 1 above)
Comment (robabram) [reply tests/helpers/unittest_base.py]: It's by accident.  Thank you for catching it.

**Review by wangy70 — APPROVED** [review]

**Review by pegbertsch — APPROVED** [review]

**Pull request merged**